poppy/wfe.py

```diff
@@ -353,6 +353,151 @@ def get_opd(self, wave):
         phase_screen = np.fft.ifftshift(np.fft.ifft2(np.fft.ifftshift(scaled))).real   # FT of scaled random PSD makes phase screen
 
         phase_screen -= np.mean(phase_screen)  # force zero-mean
-        opd = phase_screen / np.std(phase_screen) * self.wfe.to(u.m).value  # normalize to wanted input rms wfe
+        self.opd = phase_screen / np.std(phase_screen) * self.wfe.to(u.m).value  # normalize to wanted input rms wfe
 
-        return opd
+        return self.opd
+
+
+class PowerSpectrumWFE(WavefrontError):
+    """
+    Power spectrum PSD WFE class from characterizing and modeling 
+    optical surface power spectrum and applying optical noise.
+    
+    References:
+    Males, Jared. MagAO-X Preliminary-Design Review, 
+        Section 5.1: Optics Specifications, Eqn 1
+        https://magao-x.org/docs/handbook/appendices/pdr/
+    Lumbres, et al. In Prep.
+
+    Parameters
+    ----------
+    name : string
+        name of the optic
+    psd_parameters: list of array with various astropy quantities
+        Specifies the various PSD parameters with appropriate units, 
+        ordered according to which PSD model set. Follows the order:
+        [alpha, beta, outer_scale, inner_scale, surf_roughness]
+    psd_weight: iterable list of floats
+        Specifies the weight muliplier to set onto each model PSD
+    seed : integer
+        Seed for the random phase screen generator
+    apply_reflection: boolean
+        Applies 1/2 scale for the OPD as needed for reflection
+        Set to True if the PSD model does not account already for reflection.
+    screen_size: integer
+        Sets how large the PSD matrix will be calculated.
+        If None passed in, then code will default size to 4x wavefront's side.
```

Comment:
Consider adding here some brief explanation for why the user might want to adjust this, and what effect it has. It's probably not obvious to a non-expert why it's desirable to make this much larger than the wavefront.

Reply:
Okay. I have edited this in the comment block. I think my reasoning is sufficient? It could use more detail, maybe.